### PR TITLE
Host IR: add `GetCurrentStream`

### PR DESCRIPTION
**Pull request by samnordmann**
# What

adds the primitive `GetCurrentStream` to Host Ir stack.

# Why

needed for 
- https://github.com/NVIDIA/Fuser/pull/3606

The idea is that if we want to use multiple stream internally, we need before hand to capture the user stream and to set it back to being the active stream when returning

### CODE REVIEW

**Comment by samnordmann** [comment]
!test

**Comment by samnordmann** [comment]
!test

File: csrc/host_ir/host_ir.cpp [hunk 1]
```diff
@@ -179,6 +179,32 @@ bool SetCurrentStream::sameAs(const Statement* other) const {
...
+// TODO: implement better ?
+std::string GetCurrentStream::toInlineString(int indent_size) const {
+  NVF_CHECK(false, "Cannot be printed inline");
+}
+
+// TODO: implement
+bool GetCurrentStream::sameAs(const Statement* other) const {
+  return false;
+}
+
```
Comment (wujingyue) [inline csrc/host_ir/host_ir.cpp:207]:
```suggestion
```
How about just the defaults? https://github.com/NVIDIA/Fuser/blob/56ad61c29fa65830540c4644be8a3b4da02533ce/csrc/ir/base_nodes.cpp#L324 and https://github.com/NVIDIA/Fuser/blob/56ad61c29fa65830540c4644be8a3b4da02533ce/csrc/ir/base_nodes.cpp#L67

File: csrc/host_ir/executor.cpp [hunk 2]
```diff
@@ -273,6 +273,14 @@ void HostIrEvaluator::handle(SetCurrentStream* set_current_stream) {
   setCurrentCUDAStream(getCUDAStream(set_current_stream->stream()));
 }
 
+void HostIrEvaluator::handle(GetCurrentStream* get_current_stream) {
+  auto my_device_index = static_cast<c10::DeviceIndex>(
+      communicator_ ? communicator_->deviceId() : 0);
```
Comment (wujingyue) [inline csrc/host_ir/executor.cpp:278]: Your other PR stores my_device_index_ so you could use it here to save a little typing.

**Review by wujingyue — APPROVED** [review]

File: csrc/host_ir/host_ir.cpp (hunk 1 above)
Comment (samnordmann) [reply csrc/host_ir/host_ir.cpp:207]: ok

**Comment by samnordmann** [comment]
!test

**Pull request merged**